packages/jest-mock/src/index.ts

```diff
@@ -973,7 +985,7 @@ class ModuleMockerClass {
     return object[methodName];
   }
 
-  _spyOnProperty<T extends {}, M extends keyof T>(
+  private _spyOnProperty<T extends {}, M extends keyof T>(
```

Comment:
```suggestion
  private _spyOnProperty<T extends {}, M extends NonFunctionPropertyNames<T>>(
```

?

Reply:
This causes TS to choke :( Wanna have a look locally?

Reply:
https://github.com/facebook/jest/pull/7850/commits/c12feb3bc2c8cbd24455c97df4c6d48c9e262643